### PR TITLE
Post MMU/MemInterface merge cleanup

### PR DESCRIPTION
**Pull request by rahahahat**
Includes:
- Ports are now shared_ptrs
- Size change of `size` in MemoryAcessTarget
- Variable name change to vaddr in MemoryAccessTarget
- Removed unnecessary moves of unique_ptr to references
- General refactor of functions

### CODE REVIEW

File: src/include/simeng/Port.hh [hunk 1]
```diff
@@ -63,16 +60,15 @@ class Port {
...
-  Port() : id_(idctr_++) {}
+  Port() {}
 
  private:
-  /** Order of a port used to convey the order in which the ports are connected,
-   * this is used to determine the destination of each port. */
-  uint8_t order_ = 0;
+  /** Id of a port local to a port mediator connection. This is mainly used to
+   * find destination ports for source ports present inside a mediator. */
+  uint16_t local_id_;
```
Comment (FinnWilkinson) [inline src/include/simeng/Port.hh:71]: Could you re-word this comment as it isn't that clear

File: src/include/simeng/memory/MemPacket.hh [hunk 2]
```diff
@@ -87,13 +89,13 @@ class MemPacket {
...
-                                                      uint64_t size,
+  static std::unique_ptr<MemPacket> createReadRequest(uint64_t address,
+                                                      uint16_t size,
                                                       uint64_t reqId);
 
   /** Static function used to create a write request. */
-  static std::unique_ptr<MemPacket> createWriteRequest(uint64_t vaddr,
-                                                       uint64_t size,
+  static std::unique_ptr<MemPacket> createWriteRequest(uint64_t address,
+                                                       uint16_t size,
```
Comment (FinnWilkinson) [inline src/include/simeng/memory/MemPacket.hh:98]: Sould probably specify if paddr or vaddr for createRequest

File: src/include/simeng/memory/MemPacket.hh [hunk 3]
```diff
@@ -117,19 +119,20 @@ class MemPacket {
...
-  std::vector<char> data_;
+  /** Payload assosciate with a MemPacket. */
+  std::vector<char> payload_;
 
   /** Default constructor of a MemPacket. */
   MemPacket() {}
 
   /** Constructor for MemPackets which do not hold any data. */
-  MemPacket(uint64_t vaddr, uint64_t size, MemPacketType type, uint64_t reqId);
+  MemPacket(uint64_t address, uint16_t size, MemPacketType type,
```
Comment (FinnWilkinson) [inline src/include/simeng/memory/MemPacket.hh:129]: Specify if paddr or vaddr

File: src/include/simeng/memory/MemPacket.hh [hunk 4]
```diff
@@ -117,19 +119,20 @@ class MemPacket {
...
+            uint64_t reqId);
 
   /** Constructor for MemPackets which hold any data. */
-  MemPacket(uint64_t vaddr, uint64_t size, MemPacketType type, uint64_t reqId,
+  MemPacket(uint64_t address, uint16_t size, MemPacketType type, uint64_t reqId,
```
Comment (FinnWilkinson) [inline src/include/simeng/memory/MemPacket.hh:133]: Specify if paddr or vaddr

File: src/include/simeng/memory/MemPacket.hh (hunk 2 above)
Comment (rahahahat) [reply src/include/simeng/memory/MemPacket.hh:98]: done

File: src/include/simeng/memory/MemPacket.hh (hunk 3 above)
Comment (rahahahat) [reply src/include/simeng/memory/MemPacket.hh:129]: done

File: src/include/simeng/memory/MemPacket.hh (hunk 4 above)
Comment (rahahahat) [reply src/include/simeng/memory/MemPacket.hh:133]: done

**Pull request merged**